File: 0162-find-peak-element/0162-find-peak-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {

    let length = nums.length;

    if(length === 1 || nums[0] > nums[1])
        return 0;
    
    if(nums[length - 1] > nums[length - 2])
        return length - 1;
    
    let low = 1;
    let high = length - 2;

    while(low <= high)
    {
        let mid = Math.trunc((low + high) / 2);

        if(nums[mid] > nums[mid + 1] && nums[mid] > nums[mid - 1])
            return mid;
        else if(nums[mid] > nums[mid - 1])
            low = mid + 1;
        else
            high = mid - 1;
    }

    return -1;
};

File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {

        int length = nums.length;

        if(length == 1)
            return 0;

        if(nums[0] > nums[1])
            return 0;
        
        if(nums[length - 1] > nums[length - 2])     
            return length - 1;
        
        int low = 1;
        int high = length - 2;

        while(low <= high)
        {
            int mid = low + (high - low) / 2;

            if(nums[mid] > nums[mid - 1] && nums[mid] > nums[mid + 1])
                return mid;
            else if(nums[mid - 1] <= nums[mid])
                low = mid + 1;
            else
                high = mid - 1;
        }

        return -1;
    }
}